packages/bitcore-lib/test/script/interpreter.js

```diff
@@ -443,18 +417,99 @@ describe('Interpreter', function() {
             if (!verified) {
               allInputsVerified = false;
             }
-          });
+          }
           var txVerified = tx.verify();
           txVerified = (txVerified === true) ? true : false;
           allInputsVerified = allInputsVerified && txVerified;
           allInputsVerified.should.equal(expected);
-
         });
-      });
+      }
     };
     test_txs(tx_valid, true);
     test_txs(tx_invalid, false);
 
   });
 
+
+  const allConsensusFlags = function() {
+    const ret = [];
+    for (let i = 0; i < 128; ++i) {
+      let flag = 0;
+      if (i & 1) flag |= Interpreter.SCRIPT_VERIFY_P2SH;
+      if (i & 2) flag |= Interpreter.SCRIPT_VERIFY_DERSIG;
+      if (i & 4) flag |= Interpreter.SCRIPT_VERIFY_NULLDUMMY;
+      if (i & 8) flag |= Interpreter.SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY;
+      if (i & 16) flag |= Interpreter.SCRIPT_VERIFY_CHECKSEQUENCEVERIFY;
+      if (i & 32) flag |= Interpreter.SCRIPT_VERIFY_WITNESS;
+      if (i & 64) flag |= Interpreter.SCRIPT_VERIFY_TAPROOT;
+
+      // SCRIPT_VERIFY_WITNESS requires SCRIPT_VERIFY_P2SH
+      if (flag & Interpreter.SCRIPT_VERIFY_WITNESS && !(flag & Interpreter.SCRIPT_VERIFY_P2SH)) continue;
+      // SCRIPT_VERIFY_TAPROOT requires SCRIPT_VERIFY_WITNESS
+      if (flag & Interpreter.SCRIPT_VERIFY_TAPROOT && !(flag & Interpreter.SCRIPT_VERIFY_WITNESS)) continue;
+
+      ret.push(flag);
+    }
+
+    return ret;
+  }
+
+  describe('bitcoind script asset tests', function() {
+    this.timeout(60000); // Some tests run a little long
+
+    for (let i = 0; i < script_asset_tests.length; i++) {
+      let test = script_asset_tests[i];
+      it(`script asset test vector ${i}: ${test.comment}`, function() {
+        const tx =  new Transaction(test.tx);
+        const prevOuts = [];
+        for (let prevout in test.prevouts) {
```

Comment:
should be for of.

Reply:
I reverted this back to a for..in loop since the index is needed in the test.